### DISCUSSION TITLE
Having issues with put/post in a Python project

### DISCUSSION THREAD

**Discussion by c9victor** [coddingtonbear/obsidian-local-rest-api#99] [Q&A] [answered]
Hello! I am not very used to making API calls and haven't done a lot of projects involving them. I am running into some slight issues and was hoping someone might have answers because Chat GPT has not had anything useful to say to me. 

I have tried creating new obsidian pages using both POST and PUT requests, but when the pages are created they are _empty_ text files instead of Markdown files with the information that I am looking for. As you can see in the screenshots, the text variable is not empty, but the information is not getting appended to the files. Please help me understand where I am going wrong.

![obsidian api proj](https://github.com/coddingtonbear/obsidian-local-rest-api/assets/56898542/7f962ff9-bca8-4fde-9d38-4a6942486281)
![obsidian api proj2](https://github.com/coddingtonbear/obsidian-local-rest-api/assets/56898542/aa4f8c51-edd0-4144-b605-c6f2e1567d6b)

**Comment by coddingtonbear** [comment]
It's a little difficult for me to follow what you're expecting here, but a `PUT` request in a REST API like this does not traditionally return the document back to you.  If you want to see what the document looks like after you perform such a request, you can just follow it with a `GET` request, though... or just look at the payload content you're sending to the API, of course.

**Reply by c9victor** [reply, marked as answer]
Thank you for the response. I finally figured out that it was because there are colons in the title of the obsidian pages I am trying to create  (e.g., "Mark 1:1"). I can get around it by simply stripping out the colons for now. I would _like_ to keep the colons, and have tried to encode them as done [here](https://stackoverflow.com/questions/42349000/python3-requests-use-quote-instead-of-quote-plus) and by doing a replace like so: reference.replace(':', ' ')

But, neither option worked for me. Thank you for your time and this fun API that you have created.

**Reply by coddingtonbear** [reply]
Unfortunately, Obsidian won't let you use a colon in a page title because that's not supported in filenames on some filesystems that Obsidian runs on.  So I'm afraid you won't have much luck around that particular challenge.
